lib/recog/nizer.rb

```diff
@@ -41,6 +41,24 @@ def self.match(match_key, match_string)
     nil
   end
 
+  def self.multi_match(match_key, match_string)
```

Comment:
I would suggest just adding this on to `self.match` much in the same way you changed initialize in `lib/recog/matcher.rb`, except this time make it take an optional, off-by default third argument to turn multi-match on/off.

Metasploit uses Recog::Nizer.match to do most of its work and it would be nice to just have one method, not two.